xgboost_ray/sklearn.py

```diff
@@ -0,0 +1,682 @@
+        iteration_range=None,
+        ray_params: Union[None, RayParams, Dict] = None,
+        _remote: Optional[bool] = None,
+):
+    iteration_range = _convert_ntree_limit(model.get_booster(), ntree_limit,
+                                           iteration_range)
+    iteration_range = model._get_iteration_range(iteration_range)
+
+    ray_params = _set_ray_params_n_jobs(ray_params, model.n_jobs)
+
+    test = RayDMatrix(X, base_margin=base_margin, missing=model.missing)
+    return predict(
+        model.get_booster(),
+        data=test,
+        iteration_range=iteration_range,
+        output_margin=output_margin,
+        validate_features=validate_features,
+        ray_params=ray_params,
+        _remote=_remote,
+    )
+
+
+class RayXGBRegressor(XGBRegressor):
+    @_deprecate_positional_args
+    def fit(self,
+            X,
+            y,
+            *,
+            sample_weight=None,
+            base_margin=None,
+            eval_set=None,
+            eval_metric=None,
+            early_stopping_rounds=None,
+            verbose=True,
+            xgb_model: Optional[Union[Booster, str, "XGBModel"]] = None,
+            sample_weight_eval_set=None,
+            base_margin_eval_set=None,
+            feature_weights=None,
+            callbacks=None,
+            ray_params: Union[None, RayParams, Dict] = None,
+            _remote: Optional[bool] = None):
+        evals_result = {}
+
+        # enable_categorical param has been added in xgboost 1.5.0
+        try:
+            train_dmatrix, evals = _wrap_evaluation_matrices(
+                missing=self.missing,
+                X=X,
+                y=y,
+                group=None,
+                qid=None,
+                sample_weight=sample_weight,
+                base_margin=base_margin,
+                feature_weights=feature_weights,
+                eval_set=eval_set,
+                sample_weight_eval_set=sample_weight_eval_set,
+                base_margin_eval_set=base_margin_eval_set,
+                eval_group=None,
+                eval_qid=None,
+                create_dmatrix=lambda **kwargs: RayDMatrix(**kwargs),
+                enable_categorical=self.enable_categorical,
+            )
+        except AttributeError as e:
+            if "enable_categorical" not in str(e):
+                raise e
+            train_dmatrix, evals = _wrap_evaluation_matrices(
+                missing=self.missing,
+                X=X,
+                y=y,
+                group=None,
+                qid=None,
+                sample_weight=sample_weight,
+                base_margin=base_margin,
+                feature_weights=feature_weights,
+                eval_set=eval_set,
+                sample_weight_eval_set=sample_weight_eval_set,
+                base_margin_eval_set=base_margin_eval_set,
+                eval_group=None,
+                eval_qid=None,
+                create_dmatrix=lambda **kwargs: RayDMatrix(**kwargs),
+            )
```

Comment:
This code block seems to be in the other regressors as well. Does it make sense to have a `RayScikitLearnBase` (like Dask does) that implements some of this logic for all regressors?

Reply:
XGBoost sklearn API has some funky way to dynamically generate `get_params` that breaks when introducing new classes to the inheritance. I suppose that doing something similar to xgboost-dask sklearn interface could work here. My reasoning was that by inheriting from the estimators themselves we would avoid breaking any `isinstance` checks in user code, making this 100% drag-and-drop replace, but you are right in saying that a base class would provide more flexibility.